app/services/scholars_archive/google_scholar_citation_service.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module ScholarsArchive
+  # Class to handle the expected format for citation publication dates as
+  # required in the indexing guidelines for configuring meta-tags:
+  # https://scholar.google.com/intl/en/scholar/inclusion.html#indexing
+  class GoogleScholarCitationService
+    attr_reader :input_date
+
+    def self.call(input_date)
+      new(input_date).call
+    end
+
+    def call
+      format_date
+    end
+
+    private
+
+    def initialize(input_date)
+      @edtf_value = edtf_date(input_date)
+    end
+
+    def edtf_date(date)
+      edtf_val = Date.edtf(date)
+      if edtf_val.blank? && date.present?
+        # date_uploaded usually comes in the form m/d/Y unlike edtf that comes in the form Y-m-d
+        custom_date = Date.strptime(date, '%m/%d/%Y')
+        edtf_val = Date.edtf(custom_date.to_s)
+      end
+      edtf_val
+    rescue StandardError => e
+      Rails.logger.warn "GoogleScholarCitationService #{e.message}"
+      nil
```

Comment:
Is this improperly indented?

Reply:
It's intended to catch errors like invalid inputs (and return `nil` so that if a date is invalid, it may proceed with an empty value without breaking the show page). `Date` may fail with `ArgumentError: invalid date`. See spec https://github.com/osulp/Scholars-Archive/blob/google-scholar-citation-date/spec/services/scholars_archive/google_scholar_citation_service_spec.rb#L19